lib/Doctrine/Migrations/Version/Factory.php

```diff
@@ -14,25 +16,23 @@
  */
 class Factory
 {
-    /** @var Configuration */
-    private $configuration;
+    /** @var Connection */
+    private $connection;
 
-    /** @var ExecutorInterface */
-    private $versionExecutor;
+    /** @var LoggerInterface */
+    private $logger;
 
-    public function __construct(Configuration $configuration, ExecutorInterface $versionExecutor)
+    public function __construct(Connection $connection, LoggerInterface $logger)
     {
-        $this->configuration   = $configuration;
-        $this->versionExecutor = $versionExecutor;
+        $this->connection = $connection;
+        $this->logger     = $logger;
     }
 
-    public function createVersion(string $version, string $migrationClassName) : Version
+    public function createVersion(string $migrationClassName) : AbstractMigration
```

Comment:
I'm confused: this class is in the `Version` namespace but returns a migration. Should we rename it `MigrationFactory` instead of just `Factory`?